sway/desktop/xdg_shell.c

```diff
@@ -287,10 +287,8 @@ static void handle_commit(struct wl_listener *listener, void *data) {
 
 	struct wlr_box new_geo;
 	wlr_xdg_surface_get_geometry(xdg_surface, &new_geo);
-	bool new_size = new_geo.width != view->geometry.width ||
-			new_geo.height != view->geometry.height ||
-			new_geo.x != view->geometry.x ||
-			new_geo.y != view->geometry.y;
+	bool new_size = memcmp(&new_geo, &view->geometry,
```

Comment:
Hm, I'm concerned about padding. The struct _probably_ won't have padding, but `memcmp` would make it extremely difficult to debug if it ever has. Maybe introduce a helper for this?

Reply:
Ugh, you're right, undid the change. :|